xchange-blockchain/src/test/java/org/knowm/xchange/blockchain/service/trade/TradeServiceTest.java

```diff
@@ -0,0 +1,154 @@
+import org.junit.Test;
+import org.knowm.xchange.blockchain.BlockchainExchange;
+import org.knowm.xchange.blockchain.service.BlockchainBaseTest;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.trade.*;
+import org.knowm.xchange.service.trade.TradeService;
+import org.knowm.xchange.service.trade.params.CancelOrderByCurrencyPair;
+import org.knowm.xchange.service.trade.params.TradeHistoryParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.TradeHistoryParams;
+import org.knowm.xchange.service.trade.params.TradeHistoryParamsTimeSpan;
+import si.mazi.rescu.HttpStatusIOException;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.Date;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.knowm.xchange.blockchain.service.utils.BlockchainConstants.*;
+
+public class TradeServiceTest extends BlockchainBaseTest {
+    private TradeService service;
+
+    @Before
+    public void init() {
+        BlockchainExchange exchange = createExchange();
+        service = exchange.getTradeService();
+    }
+
+    @Test(timeout = 2000)
+    public void getOpenOrdersSuccess() throws Exception {
+        stubGet(ORDERS_JSON, 200, URL_ORDERS);
+        OpenOrders response = service.getOpenOrders();
+        Assert.assertNotNull(response);
+    }
+
+    @Test(timeout = 2000)
+    public void placeLimitOrderSuccess() throws Exception {
+        String response = placeLimitOrder();
+        Assert.assertNotNull(response);
+    }
+
+    @Test(timeout = 2000)
+    public void placeMarketOrderSuccess() throws Exception {
+        String response = placeMarketOrder();
+        Assert.assertNotNull(response);
+    }
+
+    @Test(timeout = 2000)
+    public void placeStopOrderSuccess() throws Exception {
+        String response = placeStopOrder();
+        Assert.assertNotNull(response);
+    }
+
+    @Test(timeout = 2000)
+    public void cancelOrderSuccess() throws Exception {
+        Boolean response = cancelOrder(200, ORDER_ID);
+        assertThat(response).isEqualTo(true);
+    }
+
+    @Test(timeout = 2000)
+    public void cancelOrderFailure() throws Exception {
+        Throwable exception = catchThrowable(() -> cancelOrder(400, ORDER_ID));
+        assertThat(exception)
+                .isInstanceOf(HttpStatusIOException.class)
+                .hasMessage(HTTP_CODE_400);
+    }
+
+    @Test(timeout = 2000)
+    public void cancelOrderByCurrency() throws Exception {
+        CancelOrderByCurrencyPair cancelOrderByCurrencyPair = () -> new CurrencyPair("BTC/USD");
+        Boolean response = cancelAllOrder(200, cancelOrderByCurrencyPair);
+        assertThat(response).isEqualTo(true);
+    }
+
+    @Test(timeout = 2000)
+    public void getTrades() throws Exception {
+        TradeHistoryParams params = service.createTradeHistoryParams();
```

Comment:
Make your service returns BlockchainTradeHistoryParams so you do not need all this casting.